File: 2870-minimum-number-of-operations-to-make-array-empty/2870-minimum-number-of-operations-to-make-array-empty.py
class Solution:
    def minOperations(self, nums: List[int]) -> int:
        map = defaultdict(int)
        for num in nums:
            map[num] += 1
        
        min = 0
        for val in map.values():
            if val == 1:
                return -1
            op = math.floor(val/3)
            rem = val % 3
            if rem > 0:
                op += 1
            min += op
        
        return min
        
        

File: 2870-minimum-number-of-operations-to-make-array-empty/2870-minimum-number-of-operations-to-make-array-empty.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var minOperations = function(nums) {
    let map = new Map();
    for(let i = 0; i < nums.length; i++){
        map.set(nums[i], map.get(nums[i]) + 1 || 1)
    }
    let min = 0;
    
    for(let val of map.values()){
        if(val === 1) {
            return -1;
        }
        let vr = Math.floor(val / 3);
        let rem = val % 3;
        if(rem > 0) {
            vr++;
        }
      min += vr;
    }
    return min
};